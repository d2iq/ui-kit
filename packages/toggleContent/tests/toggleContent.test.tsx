import React from "react";
import { createSerializer } from "@emotion/jest";
import { render } from "@testing-library/react";

import { ToggleContent } from "..";

expect.addSnapshotSerializer(createSerializer());

const primary = () => <div>primary</div>;
const secondary = () => <div>secondary</div>;

describe("ToggleContent", () => {
  beforeEach(() => {
    const mock = jest.fn();
    mock.mockReturnValue({
      toString: jest.fn()
    });
    window.getSelection = mock;
  });

  it("renders a div wrapping the content passed as props", () => {
    const { asFragment } = render(
      <ToggleContent tabIndex="0" contentOn="Hello" contentOff="Bye" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("string content", () => {
    const { getByTestId } = render(
      <ToggleContent contentOn="Hello" contentOff="Bye" />
    );
    const toggleElement = getByTestId("toggleContent");

    expect(toggleElement).toHaveTextContent("Hello");
    toggleElement.click();
    expect(toggleElement).toHaveTextContent("Bye");
  });

  it("component content", () => {
    const { getByTestId } = render(
      <ToggleContent contentOn={primary()} contentOff={secondary()} />
    );
    const toggleElement = getByTestId("toggleContent");

    expect(toggleElement).toHaveTextContent("primary");
    toggleElement.click();
    expect(toggleElement).toHaveTextContent("secondary");
  });
});
