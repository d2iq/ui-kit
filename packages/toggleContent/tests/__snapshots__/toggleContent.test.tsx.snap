// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleContent renders a div wrapping the content passed as props 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-bottom: 1px dashed;
  cursor: pointer;
  padding-bottom: 1px;
  cursor: pointer;
}

<span
    class="emotion-0"
    data-cy="toggleContent"
    role="button"
    tabindex="0"
  >
    Hello
  </span>
</DocumentFragment>
`;
